src/test_runner/TestResult.ts

```diff
@@ -1,19 +1,25 @@
+import TestStatus from './TestStatus';
+
 /**
- * Indicates what the result of a test run was.
+ * Indicates the result of single test
  */
-enum TestResult {
+interface TestResult {
+  /**
+   * The full human readable name of the test
+   */
+  name: string;
   /**
-   * The TestRunner was able to complete the run. NOTE: This can still mean that one or more tests failed.
+   * The status of the test
    */
-  Complete,
+  status: TestStatus;
   /**
-   * The TestRunner was unable to complete the run due to an unexpected error. For example: the code contains a syntax error
+   * Optional: any error messages
    */
-  Error,
+  errorMessages?: string[];
```

Comment:
Ah no. I was thinking that the `errorMessages` in RunResult would be error messages to do with the test run itself. The `errorMessages` in `TestResult` would have to do with errors during the run of that test. 

Question is, do we need `errorMessages` on TestResult? I cannot think of a use case for it right this second. Maybe we could rename it to be `failedMessages`? And fill it with messages when a test fails? 

Will fix bithound 👍